src/net.c

```diff
@@ -457,6 +471,41 @@ coap_new_context(
 
   memset(c, 0, sizeof(coap_context_t));
 
+#ifdef HAVE_EPOLL_SUPPORT
+  c->epfd = epoll_create1(0);
+  if (c->epfd == -1) {
+    coap_log(LOG_ERR, "coap_new_context: Unable to epoll_create: %s (%d)\n",
+             coap_socket_strerror(),
+             errno);
+  }
+  if (c->epfd != -1) {
+    c->eptimerfd = timerfd_create(CLOCK_REALTIME, TFD_NONBLOCK);
```

Comment:
I am not sure if you can always rely on CLOCK_REALTIME to be available. In coap_time.c, after some testing, I found this condition for `clock_gettime()`: 

     #if _POSIX_TIMERS && !defined(__APPLE__)
     #define COAP_CLOCK CLOCK_REALTIME
     #endif

Reply:
Updated code for consistency.  However as epoll is really only supported by Linux, I doubt that there would be other suitable alternative clocks for `timerfd_create(`).

Reply:
Code does not compile as is - I need to put in the _POSIX_TIMERS wrapper (which really is for colck_gettime()) so have reverted code back to CLOCK_REALTIME as this code section is in a HAVE_EPOLL_SUPPORT wrapper.  OK?

Reply:
Okay. As the epoll headers seem to be missing on Darwin, I suppose there is no point in this special treatment.